.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,39 @@
+name: "Unit Tests"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [ "3.9", "3.10", "3.11", "3.12" ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

Comment:
this is on version 5 now. I think checkout above is on 4.

Reply:
thanks!